src/Orleans.Core/Configuration/OptionLogger/IOptionsLogger.cs

```diff
@@ -0,0 +1,84 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+
+namespace Orleans
+{ 
+    internal class ClientOptionsLogger : OptionsLogger, ILifecycleParticipant<IClusterClientLifecycle>
+    {
+        private int ClientOptionLoggerLifeCycleRing = 1000;
+        public ClientOptionsLogger(ILogger<ClientOptionsLogger> logger, IServiceProvider services)
+            : base(logger, services)
+        {
+        }
+
+        public void Participate(IClusterClientLifecycle lifecycle)
+        {
+            lifecycle.Subscribe(ClientOptionLoggerLifeCycleRing, this.OnRuntimeInitializeStart);
+        }
+
+        public Task OnRuntimeInitializeStart(CancellationToken token)
+        {
+            this.LogOptions();
+            return Task.CompletedTask;
+        }
+    }
+
+    public abstract class OptionsLogger 
+    {
+        private ILogger logger;
+        private IServiceProvider services;
+        protected OptionsLogger(ILogger logger, IServiceProvider services)
+        {
+            this.logger = logger;
+            this.services = services;
+        }
+        public void LogOptions()
+        {
+            this.logger.LogInformation("Starting with following configuration:");
+            var optionFormatters = services.GetServices<IOptionFormatter>();
+            foreach (var formatterGroup in optionFormatters
+                .GroupBy(optionFormatter => optionFormatter.Category)
+                .OrderBy(group => group.Key))
+            {
+                int spaceNum = 0;
+                //log out category if there's one
+                if (!string.IsNullOrEmpty(formatterGroup.Key))
+                {
+                    this.logger.LogInformation($"{CreateNSpace(spaceNum)}{formatterGroup.Key}:");
+                    spaceNum += 2;
+                }
+
+                foreach (var optionFormatter in formatterGroup)
+                {
+                    LogOption(optionFormatter, spaceNum);
+                }
+
+            }
+        }
+
+        private void LogOption(IOptionFormatter formatter, int spaces)
```

Comment:
that would be a bit chatty, but yeah I see your point. but then I think this might be better:  logging each option in one message. So the SINGLE log message will be like:
 Category: xxx
Name: xxx
key =value,
key = value. 

This will keep information about one option in a single message, which improves readability for both console and logs.  If we spread each format line into different log messages, I don't think that will be a good experience for either console or  log.

Reply:
category is intended mostly for named option. So for example if user register multiple storage provider, then all those provider options will belong to 'storage' category. So they will be logged next to each other , or at least have the option to . Basically for readability and backward compatability. Since that is how we used to log configurations.

Reply:
log one format line per messages seems too shattering for me. for example, user will need to look up multiple log messages to get information about one option. This seems odd.

Reply:
Ok with having all fields in an option be in one line.  I just don't think that is best.  I agree it's debatable.  I am of the opinion that people are usually interested in a specific setting rather than all the settings in an option, but again.. I'm ok with either pattern.